### PR TITLE
fix(files_sharing): file request followups

### PR DESCRIPTION
**Pull request by skjnldsv**
https://github.com/user-attachments/assets/eb266847-909a-41c5-883e-1cc4c9b0e4d4

## Todo
- [x] Some design adjusting based on design team request
- [x] Simplified email validation based on groupware team feedback
- [x] Use share label as title and destination if provided
- [x] Prompt guest nickname to upload files in subfolder

## Followups
- [ ] Think about name conflicts
- [ ] Add warning when a share without READ gets change with READ
- [ ] Add time left on public page
- [ ] Add list of file requests in Navigation

### CODE REVIEW

File: apps/files_sharing/src/components/NewFileRequestDialog/NewFileRequestDialogDatePassword.vue [hunk 1]
```diff
@@ -38,15 +34,16 @@
 				:value="expirationDate"
 				name="expirationDate"
 				type="date"
-				@update:value="$emit('update:expirationDate', $event)" />
+				@input="$emit('update:expirationDate', $event)" />
+
+			<p v-if="defaultExpireDateEnforced" class="file-request-dialog__info">
+				<IconInfo :size="18" class="file-request-dialog__info-icon" />
```
Comment (susnux) [inline apps/files_sharing/src/components/NewFileRequestDialog/NewFileRequestDialogDatePassword.vue:40]: why 18 and not like everywhere 20 to be consistent?

File: apps/files_sharing/src/components/NewFileRequestDialog/NewFileRequestDialogDatePassword.vue [hunk 2]
```diff
@@ -75,20 +72,26 @@
 					</template>
 				</NcButton>
 			</div>
+
+			<p v-if="enforcePasswordForPublicLink" class="file-request-dialog__info">
+				<IconInfo :size="18" class="file-request-dialog__info-icon" />
```
Comment (susnux) [inline apps/files_sharing/src/components/NewFileRequestDialog/NewFileRequestDialogDatePassword.vue:77]: same here

**Review by susnux — APPROVED** [review]
Makes sense to me, but only checked the JS part.
2 comments but no blocker

File: apps/files_sharing/src/components/NewFileRequestDialog/NewFileRequestDialogDatePassword.vue (hunk 1 above)
Comment (susnux) [reply apps/files_sharing/src/components/NewFileRequestDialog/NewFileRequestDialogDatePassword.vue:40]: well I am fine with that then. Both icons are not too detailed so 18px does not hurt :)

File: apps/files_sharing/templates/public.php [hunk 3]
```diff
@@ -102,11 +103,11 @@
 				class="emptycontent <?php if (!empty($_['note'])) { ?>has-note<?php } ?>">
 			<?php if ($_['shareOwner']) { ?>
 				<div id="displayavatar"><div class="avatardiv"></div></div>
-				<h2><?php p($l->t('Upload files to %s', [$_['shareOwner']])) ?></h2>
-				<p><span class="icon-folder"></span> <?php p($_['filename']) ?></p>
+				<h2><?php p($l->t('Upload files to %s', [$_['label']])) ?></h2>
```
Comment (artonge) [inline apps/files_sharing/templates/public.php:106]: It is not guaranteed that label will be set, no? I would have set a fallback to filename, like below, no?

Comment (skjnldsv) [reply apps/files_sharing/templates/public.php:106]: Nice catch

File: apps/dav/lib/Files/Sharing/FilesDropPlugin.php [hunk 4]
```diff
@@ -42,25 +41,73 @@ public function enable() {
...
+			$share = $shareManager->getShareByToken($token);
+			$attributes = $share->getAttributes();
+			$isFileRequest = $attributes->getAttribute('fileRequest', 'enabled') === true;
+		} catch (\Exception $e) {
+			// Continue
+		}
+
+		// We need a valid nickname for file requests
+		if ($isFileRequest && ($nickName == null || trim($nickName) === '')) {
+			throw new MethodNotAllowed('Nickname is required for file requests');
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply apps/dav/lib/Files/Sharing/FilesDropPlugin.php:77]: No, this is a failsafe

File: apps/dav/lib/Files/Sharing/FilesDropPlugin.php [hunk 5]
```diff
@@ -42,25 +41,73 @@ public function enable() {
...
+		$isFileRequest = false;
+		$nickName = $request->getHeader('X-NC-Nickname');
+		try {
+			$shareManager = \OCP\Server::get(IManager::class);
+			$token = $this->getToken($request);
+			$share = $shareManager->getShareByToken($token);
+			$attributes = $share->getAttributes();
+			$isFileRequest = $attributes->getAttribute('fileRequest', 'enabled') === true;
+		} catch (\Exception $e) {
+			// Continue
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply apps/dav/lib/Files/Sharing/FilesDropPlugin.php:72]: We expect anything, if we're not able to get the attributes or determine the fileRequest attribute is set, then we proceed as if it is not a file request

File: apps/dav/lib/Files/Sharing/FilesDropPlugin.php [hunk 6]
```diff
@@ -42,25 +41,73 @@ public function enable() {
...
 		}
 
+		// Always upload at the root level
 		$path = explode('/', $request->getPath());
 		$path = array_pop($path);
 
+		$isFileRequest = false;
+		$nickName = $request->getHeader('X-NC-Nickname');
+		try {
+			$shareManager = \OCP\Server::get(IManager::class);
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply apps/dav/lib/Files/Sharing/FilesDropPlugin.php:66]: Not possible in plugins iirc

File: apps/files_sharing/lib/Listener/LoadPublicFileRequestAuthListener.php [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+	public function handle(Event $event): void {
+		if (!$event instanceof BeforeTemplateRenderedEvent) {
+			return;
+		}
+
+		// Make sure we are on a public page rendering
+		if ($event->getResponse()->getRenderAs() !== TemplateResponse::RENDER_AS_PUBLIC) {
+			return;
+		}
+		Util::addScript(Application::APP_ID, 'public-file-request');
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply apps/files_sharing/lib/Listener/LoadPublicFileRequestAuthListener.php:28]: Yes, I forgot I wanted to add more checks here, thanks!

File: apps/files_sharing/src/public-file-request.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+import { spawnDialog } from '@nextcloud/dialogs'
+import { defineAsyncComponent } from 'vue'
+import logger from './services/logger'
+
+const nick = localStorage.getItem('nick')
+const publicAuthPromptShown = localStorage.getItem('publicAuthPromptShown')
+
+// If we don't have a nickname or the public auth prompt hasn't been shown yet, show it
+// We still show the prompt if the user has a nickname to double check
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply apps/files_sharing/src/public-file-request.ts:14]: Currently no. We have no standard for this across apps yet.
This needs to be discussed globally I'd say

File: apps/dav/lib/Files/Sharing/FilesDropPlugin.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (artonge) [reply apps/dav/lib/Files/Sharing/FilesDropPlugin.php:72]: But should we bury the exception?

[the remark this answers is not in the document]
Comment (artonge) [reply apps/dav/lib/Files/Sharing/FilesDropPlugin.php:80]: I don't think so, if this is a file request, and something goes wrong here, then we have no information to debug why it is not behaving properly.

[the remark this answers is not in the document]
Comment (skjnldsv) [reply apps/dav/lib/Files/Sharing/FilesDropPlugin.php:80]: If something goes wrong, it means it is not a file request 😂

[the remark this answers is not in the document]
Comment (skjnldsv) [reply apps/dav/lib/Files/Sharing/FilesDropPlugin.php]: Fixed! I found a much cleaner way!!

**Review by sorbaugh — APPROVED** [review]

**Pull request merged**